src/openms/include/OpenMS/KERNEL/AnnotatedMSRawData.h

```diff
@@ -0,0 +1,153 @@
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: David Voigt $
+// $Authors: David Voigt $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/OpenMSConfig.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <boost/range/combine.hpp>
+
+#include <vector>
+
+namespace OpenMS
+{
+  class PeptideIdentification;
+  class MSSpectrum;
+
+  class OPENMS_DLLAPI AnnotatedMSRawData : public MSExperiment
+  {
+  public:
+    typedef std::pair<MSSpectrum&, std::vector<PeptideIdentification>&> Mapping;
+    typedef std::pair<const MSSpectrum&, const std::vector<PeptideIdentification>&> ConstMapping;
+
+    /// Default constructor
+    AnnotatedMSRawData() = default;
+    /// Move constructor for efficiently loading a MSExperiment without a deep copy.
+    explicit AnnotatedMSRawData(MSExperiment&& experiment) : MSExperiment(std::move(experiment)) {};
+    AnnotatedMSRawData(AnnotatedMSRawData&&) = default;
+    ~AnnotatedMSRawData() = default;
+
+    /// Get the peptide identifications for a single spectrum.
+    std::vector<PeptideIdentification>& getPeptideIds();
+    /// Get all peptide identifications for all spectra.
+    std::vector<std::vector<PeptideIdentification>>& getAllPeptideIds();
+
+    /// Set a single spectrum's peptide identification annotation
+    void setPeptideIds(std::vector<PeptideIdentification>&& ids, size_t index);
+    /// Set all peptide identifications for all spectra
+    void setAllPeptideIds(std::vector<std::vector<PeptideIdentification>>&& ids);
+
+    inline auto cbegin() const
+    {
+      return PairIterator(spectra_.cbegin(), peptide_ids.cbegin());
+    }
+
+    inline auto begin()
+    {
+      return PairIterator(spectra_.begin(), peptide_ids.begin());
+    }
+
+    inline auto end()
+    {
+      return PairIterator(spectra_.end(), peptide_ids.end());
+    }
+
+    inline auto cend() const
+    {
+      return PairIterator(spectra_.cend(), peptide_ids.cend());
+    }
+
+    inline Mapping operator[](size_t idx)
```

Comment:
I kind of don't like the accession with .first .second that much as one always needs to remember what first and second is.
what would be the advantage for the user when he mainly writes something like: 
annot_ms.getExperiment()[idx]  or annot_ms.getIdentification()[idx]
but rarely needs both